src/sentry/utils/validators.py

```diff
@@ -11,7 +11,7 @@
     "`{}` must be a valid 16 character hex (containing only digits, or a-f characters)"
 )
 
-HEXADECIMAL_16_DIGITS = re.compile("^[0-9a-fA-F]{16}$")
+HEXADECIMAL_16_DIGITS = re.compile("^[0-9a-fA-F]{1,16}$")
```

Comment:
Is this used anywhere else? Should we introduce a snuba_span_id regex, since this only applies to spans in the snuba datastore?

Reply:
So this is used by https://github.com/getsentry/sentry/blob/2cb46fe05401d7917e6a16c8963752eb95f4667d/src/sentry/utils/validators.py#L28-L29 which in turn is used by https://github.com/getsentry/sentry/blob/2cb46fe05401d7917e6a16c8963752eb95f4667d/src/sentry/api/event_search.py#L455-L464

The callers for this are
- the query builder: this is the case we're trying to fix here
- the old json query builder: this should be applicable there as well
- the endpoint for suspect spans to validate span groups: this should be applicable there as well